op-mode: T7403: add an option to forcefully remove a container image

c-po opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change summary
<!--- Provide a general summary of your changes in the Title above -->

```
If you have multiple images (tags) loaded for a container which point to the exact same hash - you can only forcefully delete them.

vyos@vyos:~$ delete container image 70dc5806
Error: unable to delete image "70dc5806" by ID with more than one tag
  ([registry.io/foo/bar:v1.0 registry.io/foo/bar:v1.0.1]):
  please force removal
```

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- optional: Link to related other tasks on Phabricator. -->
* https://vyos.dev/T7403

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->
* https://github.com/vyos/vyos-documentation/pull/1654

## How to test / Smoketest result
```
vyos@vyos:~$ add container image docker.io/jacobalberty/unifi:v8.6
vyos@vyos:~$ add container image docker.io/jacobalberty/unifi:v8.6.9
```

```
vyos@vyos:~$ delete container image 70dc580695b9
Error: unable to delete image "70dc580695b9bd2908c49f67e3100ccefd160a07164bdf475e76177e85383444" by ID with more than one tag ([docker.io/jacobalberty/unifi:v8.6.9 docker.io/jacobalberty/unifi:v8.6]): please force removal
```

```
vyos@vyos:~$ delete container image 70dc580695b9 force
vyos@vyos:~$ show container image
REPOSITORY  TAG         IMAGE ID    CREATED     SIZE
```

If an image which is still in use is about to be deleted - error out

```
vyos@vyos:~$ delete container image all force
Cannot delete image "d205499ae8bb" because it is currently being used by container "3b290f90e83d"!
```

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

sever-sever commented on src/op_mode/container.py:
```diff
@@ -72,7 +73,10 @@ def delete_image(name: str):
-    rc, output = rc_cmd(f'podman image rm {name}')
+    tmp = f'podman image rm {name}'
+    if force:
+        tmp += ' --force'
```
This option will try to stop a running container and remove the image anyway if it is used by some container.
```
vyos@r14# run show conf com | match "name c"
set container name c1 allow-host-networks
set container name c1 image 'busybox'
[edit]
vyos@r14# 
[edit]
vyos@r14# run show container image | match busy
docker.io/library/busybox                    latest        6d3e4188a38a  9 months ago  4.53 MB
[edit]
vyos@r14# 
[edit]
vyos@r14# sudo podman image rm 6d3e --force
WARN[0010] StopSignal SIGTERM failed to stop container c1 in 10 seconds, resorting to SIGKILL 
Untagged: docker.io/library/busybox:latest
Deleted: 6d3e4188a38af91b0c1577b9e88c53368926b2fe0e1fb985d6e8a70040520c4d
[edit]
vyos@r14# 
```
Though it tries to pull image again:
```
Jul 02 00:40:22 r14 systemd[1]: usr-lib-live-mount-persistence-container-storage-overlay\x2dcontainers-068c6bd35b9b8a4343fdd1384611d8c4e23a4efc08a9ee38c4339128c8023c0a-userdata-shm.mount: Deactivated successfully.
Jul 02 00:40:22 r14 systemd[1]: usr-lib-live-mount-persistence-container-storage-overlay-a3fce857461abebb28c148ec1e438a23819aacdae9025abe93b37bc55109dbd7-merged.mount: Deactivated successfully.
Jul 02 00:40:22 r14 systemd[1]: vyos-container-c1.service: Main process exited, code=exited, status=137/n/a
Jul 02 00:40:23 r14 systemd[1]: vyos-container-c1.service: Failed with result 'exit-code'.
Jul 02 00:40:23 r14 systemd[1]: vyos-container-c1.service: Scheduled restart job, restart counter is at 5.
Jul 02 00:40:23 r14 systemd[1]: Stopped VyOS Container c1.
Jul 02 00:40:23 r14 systemd[1]: Starting VyOS Container c1...
Jul 02 00:40:23 r14 podman[11581]: time="2025-07-02T00:40:23+03:00" level=warning msg="The input device is not a TTY. The --tty and --interactive flags might not work properly"
Jul 02 00:40:23 r14 podman[11581]: Resolved "busybox" as an alias (/etc/containers/registries.conf.d/shortnames.conf)
Jul 02 00:40:23 r14 podman[11581]: Trying to pull docker.io/library/busybox:latest...
Jul 02 00:40:25 r14 podman[11581]: Getting image source signatures
Jul 02 00:40:25 r14 podman[11581]: Copying blob sha256:90b9666d4aed1893ff122f238948dfd5e8efdcf6c444fe92371ea0f01750bf8c
Jul 02 00:40:25 r14 podman[11581]: Copying config sha256:6d3e4188a38af91b0c1577b9e88c53368926b2fe0e1fb985d6e8a70040520c4d
Jul 02 00:40:25 r14 podman[11581]: Writing manifest to image destination
Jul 02 00:40:25 r14 podman[11581]: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
Jul 02 00:40:25 r14 podman[11581]: ae393d1a5f2493f1fbb95801e35c02c384e521ce67d932badb6496bdf9842819
Jul 02 00:40:25 r14 systemd[1]: Started VyOS Container c1.

```
Without an Internet connection, the container will be deleted as it cannot pull the image.
Is it ok?

c-po replied:
```
cpo@LR1.wue3:~$ delete container image all force
Cannot delete image "d205499ae8bb" because it is currently being used by container "3b290f90e83d"!
```

So images that are about to be deleted but actively used will cause the delete command to fail.

c-po replied:
So images that are about to be deleted but actively used will cause the delete command to fail

c-po committed:
op-mode: T7403: podman images must be listed as root

This partially reverts commit 6b562aae2409 ("op-mode: T7459: eliminate direct
use of sudo in op mode commands") as the podman image listing currently only
works under superuser rights.

c-po committed:
op-mode: T7403: print STDOUT when adding a container image

c-po committed:
op-mode: T7403: add option for forcefully remove a container image

If you have multiple images (tags) loaded for a container which point to the
exact same hash - you can only forcefully delete them.

  vyos@vyos:~$ delete container image 70dc5806
  Error: unable to delete image "70dc5806" by ID with more than one tag
    ([registry.io/foo/bar:v1.0 registry.io/foo/bar:v1.0.1]):
    please force removal

In additon an image that is still beeing used can not be removed from the
system as this would cause infinite container restarts.

  vyos@vyos:~$ delete container image all force
  Cannot delete image "d205499ae8bb" because it is currently being used by
  container "3b290f90e83d"!

sever-sever reviewed (approved):
Add option to force delete container images.
Prevent delete images that used for the running containers.
Approved.

jestabro reviewed (approved):
Not tested, but logic appears sound.

dmbaturin reviewed (approved):
Tested and works, no objections.
